go/host/host.go

```diff
@@ -863,6 +856,92 @@ func (h *host) handleBatches(encodedBatchMsg *common.EncodedBatchMsg) error {
 	return nil
 }
 
+func (h *host) startBatchProduction() {
+	defer h.logger.Info("Stopping batch production")
+	interval := h.config.BatchInterval
+	if interval == 0 {
+		interval = 1 * time.Second
+	}
+	batchProdTicker := time.NewTicker(interval)
+	for {
+		select {
+		case <-batchProdTicker.C:
+			h.logger.Info("create batch")
+			_, err := h.enclaveClient.CreateBatch()
+			if err != nil {
+				h.logger.Warn("unable to produce batch", log.ErrKey, err)
+			}
+		case <-h.interrupter.Done():
+			return
+		}
+	}
+}
+
+func (h *host) startBatchStreaming() {
+	defer h.logger.Info("Stopping batch streaming")
+
+	// TODO: Update this to start from persisted head
+	streamChan, stop := h.enclaveClient.StreamBatches(nil)
+	var lastBatch *common.ExtBatch = nil
+	for {
+		select {
+		case resp, ok := <-streamChan:
+			if !ok {
+				stop()
+				h.logger.Warn("Batch streaming failed. Reconneting from latest received batch after 3 seconds")
+				time.Sleep(3 * time.Second)
+
+				if lastBatch != nil {
+					bHash := lastBatch.Hash()
+					streamChan, stop = h.EnclaveClient().StreamBatches(&bHash)
```

Comment:
Tiny thing but can you switch this to `h.enclaveClient` instead of the public method, probably autocomplete by accident. Caught my eye cos it's different from the one two lines below so got me wondering why.